### ISSUE TITLE
setting externalip=<version 3 onion address> results in error

### ISSUE THREAD

**Issue by Liongrass** [bitcoin/bitcoin#18254]
Hi all!

I have been running a Bitcoin node reachable via a onion address for a long time. Today I decided to upgrade to a Version 3 address, but encountered an error.

`2020-03-03T Error: Cannot resolve -externalip address: 'rn2rjwuxfjzvjfkg2hmrduc2n7tjjpasrlianfskdhhhhfqqvf7kliyd.onion'`

My new bitcoin.conf file:
```onion=127.0.0.1:9050
listen=1
externalip=rn2rjwuxfjzvjfkg2hmrduc2n7tjjpasrlianfskdhhhhfqqvf7kliyd.onion
```

My old bitcoin.conf file:
```onion=127.0.0.1:9050
listen=1
externalip=llljnatksfgtlhko.onion
```
This works fine: `2020-03-03 AddLocal(llljnatksfgtlhko.onion:8333,4)`

Bitcoin Core version:
v0.18.0.0-g2472733a24a9364e4c6233ccd04166a26a68cc65

**Comment by sipa** [comment]
Yes, Onion v3 addresses are not currently supported. A P2P protocol extension is needed to be able to relay those (see BIP 155).

**Comment by laanwj** [comment]
Correct, see issue #9214, and PR #16748

**Comment by vasild** [comment]
This is now [implemented](https://github.com/bitcoin/bitcoin/pull/19954) and will be included in `0.21`.
